catalog/transformer/catalog-transformer-xlsx/src/main/java/ddf/catalog/transformer/xlsx/XlsxMetacardUtility.java

```diff
@@ -0,0 +1,118 @@
+import javax.activation.MimeType;
+import javax.activation.MimeTypeParseException;
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.CellStyle;
+import org.apache.poi.ss.usermodel.Font;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.ss.usermodel.Sheet;
+import org.apache.poi.ss.usermodel.Workbook;
+import org.apache.poi.xssf.usermodel.XSSFWorkbook;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class XlsxMetacardUtility {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(XlsxMetacardUtility.class);
+
+  private static MimeType mimeType = new MimeType();
+
+  static {
+    try {
+      mimeType.setPrimaryType("application");
+      mimeType.setSubType("vnd.openxmlformats-officedocument.spreadsheetml.sheet");
+    } catch (MimeTypeParseException e) {
+      LOGGER.warn("Failure creating MIME type", e);
+      throw new ExceptionInInitializerError(e);
+    }
+  }
+
+  public BinaryContent buildSpreadSheet(List<Metacard> metacards) {
+
+    Set<AttributeDescriptor> attributeDescriptors =
+        metacards.get(0).getMetacardType().getAttributeDescriptors();
+
+    int rowIndex = 0;
+    int cellIndex = 0;
+
+    Workbook workbook = new XSSFWorkbook();
+    Sheet sheet = workbook.createSheet();
+    Row row = sheet.createRow(rowIndex++);
+
+    CellStyle style = workbook.createCellStyle();
+    Font font = workbook.createFont();
+    font.setBold(true);
+    style.setFont(font);
+
+    // Write header row.
+    for (AttributeDescriptor attributeDescriptor : attributeDescriptors) {
+      String attributeName = attributeDescriptor.getName();
+      Cell cell = row.createCell(cellIndex++);
+      cell.setCellValue(attributeName);
+      cell.setCellStyle(style);
+    }
+
+    for (Metacard metacard : metacards) {
+      List<String> values = getMetacardValues(attributeDescriptors, metacard);
+      row = sheet.createRow(rowIndex++);
+
+      cellIndex = 0;
+      for (String value : values) {
+        row.createCell(cellIndex++).setCellValue(value);
+      }
+    }
+
+    return writeWorkbook(workbook);
+  }
+
+  private BinaryContent writeWorkbook(Workbook workbook) {
+    ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+
+    try {
+      workbook.write(byteArrayOutputStream);
+    } catch (IOException e) {
+      LOGGER.warn("There was a problem writing the XLSX file.", e);
+    }
+
+    return new BinaryContentImpl(
+        new ByteArrayInputStream(byteArrayOutputStream.toByteArray()), mimeType);
+  }
+
+  private List<String> getMetacardValues(Set<AttributeDescriptor> attributes, Metacard metacard) {
+    List<String> values = new ArrayList<>();
+
+    for (AttributeDescriptor attributeDescriptor : attributes) {
+      String attributeName = attributeDescriptor.getName();
+      Attribute attribute = metacard.getAttribute(attributeName);
+
+      String value = attribute != null ? attribute.getValue().toString() : "";
```

Comment:
✏️ This portion may need to be a bit more complex to deal with multivalued attributes. 

```
if (attributeDescriptor.isMultivalued()) {
    // TODO create a comma separated list of values from this attribute
} else {
    String value = attribute != null ? attribute.getValue().toString() : "";
    values.add(value);
}
```